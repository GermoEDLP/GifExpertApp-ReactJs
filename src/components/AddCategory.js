import React, { useState } from "react";
import PropTypes from "prop-types";

const AddCategory = ({handleAdd}) => {
  const [inputValue, setInputValue] = useState("");
  const handleChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim() === "") {
      return;
    }
    handleAdd(inputValue);
    setInputValue("");
  };

  return (
    <form onSubmit={handleSubmit}>
      <p className="hide">{inputValue}</p>
      <input
        type="text"
        value={inputValue}
        name="category"
        placeholder="Add category"
        onChange={handleChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  handleAdd: PropTypes.func.isRequired,
};

export default AddCategory;
