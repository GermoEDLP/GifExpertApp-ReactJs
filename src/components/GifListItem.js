import React from "react";
import PropTypes from 'prop-types'


const GifListItem = ({title, url}) => {
  return (
    <div className="card animate__animated animate__bounceInUp">
      <img src={url} alt={title} title={title} />
      <p>{title}</p>
    </div>
  );
};

GifListItem.propTypes = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
};

export default GifListItem;