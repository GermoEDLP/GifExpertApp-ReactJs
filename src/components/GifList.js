import React from "react";
import useFetchGifs from "../hooks/useFetchGifs";
import GifListItem from "./GifListItem";
import PropTypes from "prop-types";

export const GifList = ({ categoria }) => {
  const { data, loading } = useFetchGifs(categoria);

  return (
    <div className="card-grid">
      <h3 className="titulo-cat">{categoria}</h3>
      {loading && (
        <p id="loadingText" className="animate__animated animate__flash">Loading...</p>
      )}
      {data.map((gif) => (
        <GifListItem key={gif.id} title={gif.title} url={gif.url} />
      ))}
    </div>
  );
};

GifList.propTypes = {
  categoria: PropTypes.string.isRequired,
};
