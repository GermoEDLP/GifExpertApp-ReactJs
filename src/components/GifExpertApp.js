import React, { useState } from "react";
import  AddCategory  from "./AddCategory";
import { GifList } from "./GifList";

export const GifExpertApp = ({defaultCategorias = []}) => {
  const [categorias, setCategorias] = useState(defaultCategorias);

  const handleAdd = (cat) => {
    setCategorias([cat, ...categorias]);
  };

  return (
    <div>
      <h2>GifExpertApp</h2>
      <hr />

      <AddCategory handleAdd={handleAdd}/>

      <ol>
        {categorias.map((categoria) => (
          <GifList categoria={categoria} key={categoria} />
        ))}
      </ol>
    </div>
  );
};
