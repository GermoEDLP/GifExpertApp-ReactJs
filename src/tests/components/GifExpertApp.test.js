import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { GifExpertApp } from "../../components/GifExpertApp";

describe("Pruebas con <GifExpertApp />", () => {
  const wrapper = shallow(<GifExpertApp />);

  test("el compoenente debe crearse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar una lista de categorias", () => {
    const categories = ["One Punch", "Samurai X", "Dragon Ball"];
    const wrapper = shallow(<GifExpertApp defaultCategorias={categories}/>);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifList").length).toBe(categories.length);
  })
});
