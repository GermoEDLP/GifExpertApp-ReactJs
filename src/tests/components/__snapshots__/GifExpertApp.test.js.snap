// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con <GifExpertApp /> debe de mostrar una lista de categorias 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    handleAdd={[Function]}
  />
  <ol>
    <GifList
      categoria="One Punch"
      key="One Punch"
    />
    <GifList
      categoria="Samurai X"
      key="Samurai X"
    />
    <GifList
      categoria="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</div>
`;

exports[`Pruebas con <GifExpertApp /> el compoenente debe crearse correctamente 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    handleAdd={[Function]}
  />
  <ol />
</div>
`;
